flashinfer/gemm.py

```diff
@@ -70,6 +70,12 @@
 DEFAULT_WORKSPACE_SIZE = 32 * 1024 * 1024
 
 
+def _match_sm_version(sm_version: str):
+    major, minor = torch.cuda.get_device_capability()
```

Comment:
Consider calling https://github.com/flashinfer-ai/flashinfer/blob/669ff33409d721a111b1b3c2ebca68ffa21e68de/flashinfer/utils.py#L212 instead, which cached this value and should be more efficient.

We observe significant CPU side overhead of the cuda runtime api to query device properties, see #1425 and #1456